src/System.Text.Primitives/System/Text/InvariantParser_decimal.cs

```diff
@@ -0,0 +1,187 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.Text.Utf8;
+
+namespace System.Text
+{
+    public static partial class InvariantParser
+    {
+        public static bool TryParse(byte[] utf8Text, int index, out decimal value, out int bytesConsumed)
+        {
+            // Precondition replacement
+            if (utf8Text.Length < 1 || index < 0 || index >= utf8Text.Length)
+            {
+                value = 0;
+                bytesConsumed = 0;
+                return false;
+            }
+
+            value = 0.0M;
+            bytesConsumed = 0;
+            string decimalString = "";
+            bool decimalPlace = false, signed = false, digitLast = false;
+
+            if (utf8Text[index] == '-' || utf8Text[index] == '+')
+            {
+                signed = true;
+                decimalString += (char)utf8Text[index];
+                index++;
+                bytesConsumed++;
+            }
+
+            for (int byteIndex = index; byteIndex < utf8Text.Length; byteIndex++)
+            {
+                byte nextByte = utf8Text[byteIndex];
+                byte nextByteVal = (byte)(nextByte - '0');
+
+                if (nextByteVal > 9)
+                {
+                    if (!decimalPlace && nextByte == '.')
+                    {
+                        if (digitLast)
+                        {
+                            digitLast = false;
+                        }
+                        bytesConsumed++;
+                        decimalPlace = true;
+                        decimalString += (char)nextByte;
+                    }
+                    else if ((decimalPlace && signed && bytesConsumed == 2) || ((signed || decimalPlace) && bytesConsumed == 1))
+                    {
+                        value = 0;
+                        bytesConsumed = 0;
+                        return false;
+                    }
+                    else
+                    {
+                        if (decimal.TryParse(decimalString, out value))
+                        {
+                            return true;
+                        }
+                        else
+                        {
+                            bytesConsumed = 0;
+                            return false;
+                        }
+                    }
+                }
+                else
+                {
+                    if (!digitLast)
```

Comment:
Ah, no, you're right -- that's an unnecessary carry-over from float/double. Changing now.